packages/schemas/alterations/next-1688375200-sync-cloud-m2m-to-logto-config.ts

```diff
@@ -0,0 +1,66 @@
+import { sql } from 'slonik';
+
+import type { AlterationScript } from '../lib/types/alteration.js';
+
+const adminTenantId = 'admin';
+
+const applicationName = 'Cloud Service';
+
+const cloudConnectionResourceIndicator = 'https://cloud.logto.io/api';
+
+enum ApplicationType {
+  Native = 'Native',
+  SPA = 'SPA',
+  Traditional = 'Traditional',
+  MachineToMachine = 'MachineToMachine',
+}
+
+const cloudConnectionConfigKey = 'cloudConnection';
+
+type Application = {
+  tenantId: string;
+  id: string;
+  name: string;
+  secret: string;
+  description: string;
+  type: ApplicationType;
+  oidcClientMetadata: unknown;
+  customClientMetadata: {
+    tenantId: string;
+  };
+  createdAt: number;
+};
+
+const alteration: AlterationScript = {
+  up: async (pool) => {
+    const rows = await pool.many<Application>(
+      sql`select * from applications where type = ${ApplicationType.MachineToMachine} and tenant_id = ${adminTenantId} and name = ${applicationName}`
+    );
+    const cloudConnections = rows.map(({ id, secret, customClientMetadata }) => ({
+      appId: id,
+      appSecret: secret,
+      tenantId: customClientMetadata.tenantId,
+    }));
+    await Promise.all(
```

Comment:
Better avoid `Promise.all` for sql queries, we can build a large sql, `insert` can have mutiline `values`

Reply:
Updated.